internal/services/storage/storage_account_resource.go

```diff
@@ -1318,6 +1319,153 @@ func resourceStorageAccountCreate(d *pluginsdk.ResourceData, meta interface{}) e
 
 	supportLevel := resolveStorageAccountServiceSupportLevel(storage.Kind(accountKind), storage.SkuTier(accountTier))
 
+	// Wait for the account services to become available
+
+	if supportLevel.supportBlob {
+		blobClient := meta.(*clients.Client).Storage.BlobServicesClient
+
+		// wait for blob service endpoint to become available
+		log.Printf("[DEBUG] waiting for %s blob service to become available", id.ID())
+		deadline, ok := ctx.Deadline()
+		if !ok {
+			return fmt.Errorf("internal-error: context had no deadline")
+		}
+		stateConf := &pluginsdk.StateChangeConf{
+			Pending: []string{"Pending"},
+			Target:  []string{"Completed"},
+			Refresh: func() (interface{}, string, error) {
+				_, err = blobClient.GetServiceProperties(ctx, id.ResourceGroupName, id.StorageAccountName)
+				if err != nil {
+					return handleStorageServiceError("blob", err)
+				}
+				return true, "Completed", nil
+			},
+			MinTimeout: 10 * time.Second,
+			Timeout:    time.Until(deadline),
+		}
+
+		if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+			return fmt.Errorf("fetching %s blob service properties: %+v", id.ID(), err)
+		}
+	}
+
+	if supportLevel.supportQueue {
+		storageClient := meta.(*clients.Client).Storage
+		account, err := storageClient.FindAccount(ctx, id.StorageAccountName)
+		if err != nil {
+			return fmt.Errorf("retrieving %s: %+v", id, err)
+		}
+		if account == nil {
+			return fmt.Errorf("unable to locate %q", id)
+		}
+
+		queueClient, err := storageClient.QueuesClient(ctx, *account)
+		if err != nil {
+			return fmt.Errorf("building Queues Client: %s", err)
+		}
+
+		// wait for queue service endpoint to become available
+		log.Printf("[DEBUG] waiting for %s queue service to become available", id.ID())
+		deadline, ok := ctx.Deadline()
+		if !ok {
+			return fmt.Errorf("internal-error: context had no deadline")
+		}
+		stateConf := &pluginsdk.StateChangeConf{
+			Pending: []string{"Pending"},
+			Target:  []string{"Completed"},
+			Refresh: func() (interface{}, string, error) {
+				properties, err := queueClient.GetServiceProperties(ctx, id.ResourceGroupName, id.StorageAccountName)
+				if err != nil {
+					return handleStorageServiceError("queue", err)
```

Comment:
I think we can simplify this to:

```suggestion
					return "Error", "Error", fmt.Errorf("waiting for the Blob %s to become available: %+v", id, err")
```

Since the `handleStorageServiceError` function below assumes that this is Completed if it's not a 404 (which would mean we mark this as Completed when it's an error?)

Reply:
If the `queueClient.GetServiceProperties` function returns error (this may be the 404 error or any other error storage API may return)

This error (generic error type) is handled by `handleStorageServiceError` function. It return `Pending`, in case of 404, however any other error is returned imidiately with `Completed` to brake the loop.

This has been requested by @magodo to avoid consuming all errors.

If i understand the logic correctly, in your example the loop will continue also in case of non 404 error?

And one exception here for queue endpoint > there is no error in case of 404 and therefore i put there the workaround below `if properties == nil`

You the owners, you can say the final word how the code should looks like. From my point of view the function `handleStorageServiceError` make sense and therefore asking for confirmation.

Reply:
@tombuildsstuff Could you please review above comment?